lib/fluent/plugin/in_udp.rb

```diff
@@ -22,12 +22,35 @@ class UdpInput < SocketUtil::BaseInput
 
     config_set_default :port, 5160
     config_param :body_size_limit, :size, default: 4096
+    config_param :delimiter, :string, :default => ""
 
     def listen(callback)
       log.info "listening udp socket on #{@bind}:#{@port}"
       @usock = SocketUtil.create_udp_socket(@bind)
       @usock.bind(@bind, @port)
+      @has_delimiter = @delimiter != ''
       SocketUtil::UdpHandler.new(@usock, log, @body_size_limit, callback)
     end
+
+    def on_message(msg, addr)
+      return super(msg, addr) unless @has_delimiter
```

Comment:
Hmm?
I think `delimiter` should be treated only as string for the facts below. 
- `delimiter` is typed as string
- developer can overwrite it with `""` in the config file
- on the other hand there is no way (or at least I cannot find out) to set `nil` from the config file